### PR TITLE
Ignore console commands that have no actions

### PR DESCRIPTION
**Pull request by alexeevdv**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #16855 

Changes summary:

- Add filter to `HelpController::getCommands` in order to ensure that command exist and it has actions. It allows to remove these checks from other places.
- `UnknownCommandException::getSuggestedAlternatives` ensure that command has default action before adding it to the suggested list.
- [Replace Nested Conditional with Guard Clauses](https://refactoring.guru/replace-nested-conditional-with-guard-clauses)

### CODE REVIEW

**Comment by alexeevdv** [comment]
Tests added

File: tests/framework/console/controllers/HelpControllerTest.php [hunk 1]
```diff
@@ -187,6 +185,37 @@ public function testActionIndexWithServeCommand()
...
+    {
+        $this->mockApplication([
+            'enableCoreCommands' => false,
+            'controllerNamespace' => 'yiiunit\data\console\controllers',
+        ]);
+        $result = Console::stripAnsiFormat($this->runControllerAction('list'));
+        $this->assertNotContains("fake-empty\n", $result);
+        $this->assertNotContains("fake-no-default\n", $result);
+        $this->assertContains("fake-no-default/index\n", $result);
+        $this->assertContains("fake\n", $result);
```
Comment (samdark) [inline tests/framework/console/controllers/HelpControllerTest.php]: There's no need to test for it. We're checking that action list doesn't contain empty controllers and nothing more.

File: tests/framework/console/controllers/HelpControllerTest.php [hunk 2]
```diff
@@ -187,6 +185,37 @@ public function testActionIndexWithServeCommand()
         $this->assertContains('--port, -p: int (defaults to 8080)', $result);
         $this->assertContains('--router, -r: string', $result);
     }
+
+    public function testCommandsFilterOnActionList()
```
Comment (samdark) [inline tests/framework/console/controllers/HelpControllerTest.php]: `testActionListContainsNoEmptyCommands`

File: tests/framework/console/controllers/HelpControllerTest.php [hunk 3]
```diff
@@ -187,6 +185,37 @@ public function testActionIndexWithServeCommand()
...
+        $result = Console::stripAnsiFormat($this->runControllerAction('list'));
+        $this->assertNotContains("fake-empty\n", $result);
+        $this->assertNotContains("fake-no-default\n", $result);
+        $this->assertContains("fake-no-default/index\n", $result);
+        $this->assertContains("fake\n", $result);
+        $this->assertContains("fake/default\n", $result);
+        $this->assertContains("fake/second\n", $result);
+    }
+
+    public function testCommandsFilterOnActionIndex()
```
Comment (samdark) [inline tests/framework/console/controllers/HelpControllerTest.php]: testActionIndexContainsNoEmptyCommands

File: tests/framework/console/controllers/HelpControllerTest.php [hunk 4]
```diff
@@ -187,6 +185,37 @@ public function testActionIndexWithServeCommand()
...
+        $this->mockApplication([
+            'enableCoreCommands' => false,
+            'controllerNamespace' => 'yiiunit\data\console\controllers',
+        ]);
+        $result = Console::stripAnsiFormat($this->runControllerAction('index'));
+        $this->assertNotContains("- fake-empty", $result);
+        $this->assertContains("- fake-no-default", $result);
+        $this->assertContains("    fake-no-default/index", $result);
+        $this->assertNotContains("    fake-no-default/index (default)", $result);
+        $this->assertContains("- fake", $result);
```
Comment (samdark) [inline tests/framework/console/controllers/HelpControllerTest.php]: There's no need to test for it. We're checking that action list doesn't contain empty controllers and nothing more.

**Review by samdark — CHANGES_REQUESTED** [review]

**Comment by samdark** [comment]
Merged with adjustments https://github.com/yiisoft/yii2/commit/6e2b1782e52f8122302b84c331b30deec8972041. Thanks for contributing.

**Pull request closed without merging**